pkgs/node/src/chain.zig

```diff
@@ -26,10 +26,17 @@ pub const BeamChain = struct {
     // from finalized onwards to recent
     states: std.AutoHashMap(types.Root, types.BeamState),
     nodeId: u32,
+    logger: *zeam_utils.ZeamLogger,
 
     const Self = @This();
-    pub fn init(allocator: Allocator, config: configs.ChainConfig, anchorState: types.BeamState, nodeId: u32) !Self {
-        const fork_choice = try fcFactory.ForkChoice.init(allocator, config, anchorState);
+    pub fn init(
+        allocator: Allocator,
+        config: configs.ChainConfig,
+        anchorState: types.BeamState,
+        nodeId: u32,
+        logger: *zeam_utils.ZeamLogger,
```

Comment:
create a params struct that you pass here and put config, anchorstate nodeid, logger into it